### ISSUE TITLE
Help with artifactory-oss installation on k3s

### ISSUE THREAD

**Issue by sultanshaikh-devops** [jfrog/charts#1885]
Hi All,

I am seeking direction and help to get artifactory-oss installation on k3s working.  Any support would be greatly appreciated.

K3s VERSION:
   v1.30.0+k3s1 (14549535)
   
export KUBECONFIG=/etc/rancher/k3s/k3s.yaml && helm install artifactory-oss \
  --set artifactory.postgresql.postgresqlPassword=password123 \
  --set artifactory.nginx.enabled=false \
  --set artifactory.postgresql.enabled=true \
  --set postgresql.enabled=true \
  --set artifactory.artifactory.service.type=traefik \
  --set artifactory.artifactory.resources.requests.cpu="500m" \
  --set artifactory.artifactory.resources.limits.cpu="4" \
  --set artifactory.artifactory.resources.requests.memory="2Gi" \
  --set artifactory.artifactory.resources.limits.memory="8Gi" \
  --set artifactory.artifactory.javaOpts.xms="1g" \
  --set artifactory.artifactory.javaOpts.xmx="3g" \
  jfrog/artifactory-oss -n artifactory-oss --create-namespace
 
 tail -f /var/lib/rancher/k3s/agent/containerd/containerd.log

time="2024-05-22T08:21:00.445568641Z" level=error msg="PullImage \"releases-docker.jfrog.io/jfrog/artifactory-oss:7.84.11\" failed" error="failed to pull and unpack image \"releases-docker.jfrog.io/jfrog/artifactory-oss:7.84.11\": failed to copy: httpReadSeeker: failed open: unexpected status code https://releases-docker.jfrog.io/v2/jfrog/artifactory-oss/blobs/sha256:7b3df940b1a034dee2310dcddba543ff75295ebc63dc66f53e65f2b5f848d5f4: **403 Forbidden"**

In addition, I am also noticing that traefik is not supported.

W0522 07:59:24.063955  230251 warnings.go:70] spec.template.spec.containers[5].ports[0]: duplicate port definition with spec.template.spec.containers[0].ports[0]
Error: INSTALLATION FAILED: 1 error occurred:
       * Service "artifactory-oss" is invalid: spec.type: Unsupported value: "traefik": supported values: "ClusterIP", "ExternalName", "LoadBalancer", "NodePort"

**Comment by amithins** [comment]
The following error is a valid one, traefik is not a supported value in service.
```
Error: INSTALLATION FAILED: 1 error occurred:
* Service "artifactory-oss" is invalid: spec.type: Unsupported value: "traefik": supported values: "ClusterIP", "ExternalName", "LoadBalancer", "NodePort"
``` 
Regarding the image pull issue on releases-docker.jfrog.io/jfrog/artifactory-oss:7.84.11, i'm able to pull it from my machine. Please check your network settings, there might be some firewall rules blocking this pull.

**Comment by sultanshaikh-devops** [comment]
> The following error is a valid one, traefik is not a supported value in service.
> 
> ```
> Error: INSTALLATION FAILED: 1 error occurred:
> * Service "artifactory-oss" is invalid: spec.type: Unsupported value: "traefik": supported values: "ClusterIP", "ExternalName", "LoadBalancer", "NodePort"
> ```
> 
> Regarding the image pull issue on releases-docker.jfrog.io/jfrog/artifactory-oss:7.84.11, i'm able to pull it from my machine. Please check your network settings, there might be some firewall rules blocking this pull.

Thank you @amithins. Let me work with our internet proxy (zscaler) team. I did validate with no proxy setting it worked fine. I will update here once I've update from our team.

In addition, I am trying to reduce number of replica by setting --set artifactory.replicaCount=1 from default 6. I am having hard time understanding why it is not applying that setting.

**Comment by amithins** [comment]
What is the version of artifactory chart ? It should be 1 by default. To explicitly set it, the following can be used,
```
--set artifactory.artifactory.replicaCount=1
```

**Comment by sultanshaikh-devops** [comment]
> What is the version of artifactory chart ? It should be 1 by default. To explicitly set it, the following can be used,
> 
> ```
> --set artifactory.artifactory.replicaCount=1
> ```

I am sorry not an expert in helm. I am assuming it is using the latest chart version 'https://charts.jfrog.io/artifactory-oss/7.18.6/'. Based on following install commands:

helm repo add jfrog https://charts.jfrog.io
helm repo update
helm install artifactory-oss jfrog/artifactory-oss -n artifactory-oss --create-namespace

Is there a way to specify specific chart version within install commands?

Thank you

**Comment by chukka** [comment]
@sultanshaikh-devops , here you go 

`helm install artifactory-oss jfrog/artifactory-oss -n artifactory-oss --create-namespace --version 107.84.11`

**Comment by sultanshaikh-devops** [comment]
@chukka thank you sir, is there a official Documation link where I can find what parameters are support with helm chart that I can use to lookup myself?  I found several links on internet but I cannot tell which one is valid.

**Comment by sultanshaikh-devops** [comment]
I am just around in circle.  Following inputs are not set, what am I missing? I've manually created pv, replicaCount does not seem to work for me as it always creates 6

--set artifactory.artifactory.replicaCount=3 
--set artifactory.postgresql.persistence.storageClassName="art-postgres-volume"


  helm upgrade --install artifactory ${CUSTOM_VALUES_FILE} \
    --namespace artifactory jfrog/artifactory-oss \
    --set artifactory.artifactory.service.type=LoadBalancer \
    --set artifactory.postgresql.postgresqlPassword=password\
    --set artifactory.nginx.enabled=false \
    --set artifactory.postgresql.enabled=true \
    --set artifactory.ingress.enabled=true \
    --set artifactory.artifactory.replicaCount=3 \
    --set artifactory.artifactory.resources.requests.cpu="1" \
    --set artifactory.artifactory.resources.limits.cpu="4" \
    --set artifactory.artifactory.resources.requests.memory="2Gi" \
    --set artifactory.artifactory.resources.limits.memory="12Gi" \
    --set artifactory.artifactory.javaOpts.xms="1g" \
    --set artifactory.artifactory.javaOpts.xmx="4g" \
    --set artifactory.postgresql.persistence.storageClassName="art-postgres-volume" \
    --set artifactory.artifactory.persistence.storageClassName="art-data-volume" \
    --version "107.84.11" --create-namespace \

**Comment by amithins** [comment]
Regarding the number 6, are you referring to the number of containers in the pod or the number of pods, can we get the output of,
```
kubectl get pods -n artifactory
```

**Comment by sultanshaikh-devops** [comment]
> Regarding the number 6, are you referring to the number of containers in the pod or the number of pods, can we get the output of,
> 
> ```
> kubectl get pods -n artifactory
> ```

Hi @amithins, below are the outputs:

1. I thought highlighted is controller by this parm --set artifactory.artifactory.replicaCount=3 

![image](https://github.com/jfrog/charts/assets/73591168/0568bf7c-4a24-420f-9ecc-753c296e17b3)

2. Second issue, I am trying to give artifactory persistence storage.  using this command --set artifactory.artifactory.persistence.storageClassName="art-data-volume". I've precreated art-data-volume as pv. 

![image](https://github.com/jfrog/charts/assets/73591168/6b823cb8-23e7-4c62-9224-3331ae25d5b6)

**Comment by sultanshaikh-devops** [comment]
> > Regarding the number 6, are you referring to the number of containers in the pod or the number of pods, can we get the output of,
> > ```
> > kubectl get pods -n artifactory
> > ```
> 
> Hi @amithins, below are the outputs:
> 
> 1. I thought highlighted is controlled by this parm --set artifactory.artifactory.replicaCount=3
> 
> ![image](https://private-user-images.githubusercontent.com/73591168/335173027-0568bf7c-4a24-420f-9ecc-753c296e17b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcwNzI5NDksIm5iZiI6MTcxNzA3MjY0OSwicGF0aCI6Ii83MzU5MTE2OC8zMzUxNzMwMjctMDU2OGJmN2MtNGEyNC00MjBmLTllY2MtNzUzYzI5NmUxN2IzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMwVDEyMzcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhMzcwYTMyYTE3ZTRmZTg0Yjg3YjdjN2FiZDFjMzJmYWQ4ZGFkYjJiODhjZGY3ZTIzMjk3NDRkYjViNGY0ZjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2w760ezehVTw0sE_FpAwvGCOs8h6ykHxSHPgUxMYKTY)
> 
> 2. Second issue, I am trying to give artifactory persistence storage.  using this command --set artifactory.artifactory.persistence.storageClassName="art-data-volume". I've precreated art-data-volume as pv.
> 
> ![image](https://private-user-images.githubusercontent.com/73591168/335174456-6b823cb8-23e7-4c62-9224-3331ae25d5b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcwNzI5NDksIm5iZiI6MTcxNzA3MjY0OSwicGF0aCI6Ii83MzU5MTE2OC8zMzUxNzQ0NTYtNmI4MjNjYjgtMjNlNy00YzYyLTkyMjQtMzMzMWFlMjVkNWI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMwVDEyMzcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhMjFhZjEyMDgzMDliYmZmMmIyOWJhNzY0MjEzZGQxNmU5ZWIwNGU4NTkwMGEyMGU3M2I5ZWZkYzQxZGFmNzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2wNUKwTbvSvXfMTY3-0iVXBwsqYZjDgehMRmturLeUk)
> 
> I can see artifactory pod has claimed the storage and it is bound. Therefore not sure why I am getting error as shown above.
> 
> ![image](https://private-user-images.githubusercontent.com/73591168/335176942-66b9a27e-f7d1-4340-822c-bdcfa5f9b47c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcwNzI5NDksIm5iZiI6MTcxNzA3MjY0OSwicGF0aCI6Ii83MzU5MTE2OC8zMzUxNzY5NDItNjZiOWEyN2UtZjdkMS00MzQwLTgyMmMtYmRjZmE1ZjliNDdjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMwVDEyMzcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3MDRlZmNlYzQ0ZTJiODVhMTRhM2VhNDQyNjA4YTk4ZThkM2JmNjU4MWNiZjgyOGIwMTUzNzQ2YjcwMWNhNGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.prLm7zNCelkRk5G58TS8KBT6WWYbfz0TZcl05LZXF0o)

Sorry I miss understood out the kubectl output. I thought ready meant replicaCount. Reading further it is number of containers within a pod.

![image](https://github.com/jfrog/charts/assets/73591168/fd1b8add-aada-453e-b7de-7d52919dd0ef)

**Comment by sultanshaikh-devops** [comment]
I've managed to fix the disk issue by manually creating in following way

apiVersion: v1
kind: PersistentVolume
metadata:
  name: art-data-volume
spec:
  accessModes:
    - ReadWriteOnce
  volumeMode: Filesystem
  persistentVolumeReclaimPolicy: Retain
  capacity:
    storage: 10Gi
  storageClassName: art-data-volume
  local:
    path: /data/art-artifactory-data
  nodeAffinity:
    required:
      nodeSelectorTerms:
      - matchExpressions:
        - key: kubernetes.io/hostname
          operator: In
          values:
          - srv01

**Now I get different issue** 

![image](https://github.com/jfrog/charts/assets/73591168/b20506a2-2510-445f-b4d3-9767dedb1ba0)


![image](https://github.com/jfrog/charts/assets/73591168/27c3f55e-dae8-4137-adca-4ac538e3c4c5)

**Comment by RobinDuhan** [comment]
Hi @sultanshaikh-devops , would you please check which container is restarting? This will be part of the `kubectl describe` output. If that doesn't provide any benefits, please share logs of the failing containers.

**Comment by sultanshaikh-devops** [comment]
> Hi @sultanshaikh-devops , would you please check which container is restarting? This will be part of the `kubectl describe` output. If that doesn't provide any benefits, please share logs of the failing containers.

Thank you, managed go past few issues. What I've noticed when I use persistent storage for postgress and antifactory.  I am seeing Master key missing, further investigation after reviewing postgress log, antifactory having issue with auth, just say's bad password.  I've confirmed manually that password is correct by loging into postgress database. Therefore, I am not sure where artifactory looks for postgress password ? 

I am passing following input during the install.

--set artifactory.postgresql.postgresqlPassword=password
